cfme/tests/automate/custom_button/test_service_vm_custom_button.py

```diff
@@ -0,0 +1,119 @@
+import fauxfactory
+import pytest
+from widgetastic_patternfly import Dropdown
+
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
+from cfme.markers.env_markers.provider import ONE
+from cfme.utils.appliance import ViaSSUI
+from cfme.utils.appliance import ViaUI
+from cfme.utils.appliance.implementations.ssui import navigate_to as ssui_nav
+from cfme.utils.appliance.implementations.ui import navigate_to as ui_nav
+
+
+pytestmark = [
+    pytest.mark.tier(2),
+    pytest.mark.provider([VMwareProvider], selector=ONE, scope="module"),
+    pytest.mark.usefixtures("setup_provider_modscope"),
+]
+
+
+@pytest.fixture(scope="module")
+def button_group(appliance):
+    collection = appliance.collections.button_groups
+    button_gp = collection.create(
+        text=fauxfactory.gen_alphanumeric(),
+        hover=fauxfactory.gen_alphanumeric(),
+        type=getattr(collection, "VM_INSTANCE"),
+    )
+    yield button_gp
+    button_gp.delete_if_exists()
+
+
+def test_custom_button_display_service_vm(request, appliance, service_vm, button_group):
+    """ Test custom button display on UI and SSUI vm resource detail page
+
+    Polarion:
+        assignee: ndhandre
+        initialEstimate: 1/2h
+        caseimportance: medium
+        caseposneg: positive
+        testtype: functional
+        startsin: 5.9
+        casecomponent: CustomButton
+        tags: custom_button
+        setup:
+            1. Add Provider
+            2. Refresh Provider; Data should be collected.
+            3. Create Simple Dialog for catalog
+                * Navigate to Automation > Automate > Customization
+                * Click on All Dialogs from a sidebar
+                * From toolbar select "Add a new Dialog"
+                * Fill Dialog's name and description
+                * Drag TextBox (we can select other as well)
+                * Save changes
+            4. Create Catalog
+                * Navigate to Services > Catalogs
+                * Click on Catalogs from a sidebar
+                * From toolbar Configuration select "Add New Catalog"
+                * Fill name and description
+                * Save changes
+            5. Create a Catalog item
+                * Navigate to Services > Catalogs
+                * From sidebar select All Catalogs > catalog (created above)
+                * From toolbar select Add New catalog item
+                * Select Provider
+                * Fill Name, description, catalog and dialog (created above)
+                * Select VM name proper template etc...
+            6. Order Catalog
+                * Navigate to Services > Catalogs
+                * Click on Service catalogs from a sidebar
+                * Order catalog from this Page
+        testSteps:
```

Comment:
I don't think its good practice to try and capture the test steps like this in automated test cases.

The level of detail is great for a manual case, but in automation the underlying methods that are achieving these actions through function calls in the test could change, without any mechanism for updating this text.  It becomes a fruitless game of trying to keep the two in sync, so that this description is accurate.

I would prefer no testSteps/expectedResults for automated tests, but am happy to allow for them.  That being said I think this goes much too far in trying to capture details that are not actually specified by the automation called here in the test.

Reply:
@mshriver I added these steps for one of the manual tests. where I tried to avoid such details and step by step explanation. Now that work helpful to me just copy and past with small changes :).

This is an unstopable debate... I would like to reduce these. Thanks :+1: